pkg/core/mpt/extension.go

```diff
@@ -47,13 +46,7 @@ func (e *ExtensionNode) Bytes() []byte {
 
 // DecodeBinary implements io.Serializable.
 func (e *ExtensionNode) DecodeBinary(r *io.BinReader) {
-	sz := r.ReadVarUint()
-	if sz > MaxKeyLength {
-		r.Err = fmt.Errorf("extension node key is too big: %d", sz)
-		return
-	}
-	e.key = make([]byte, sz)
-	r.ReadBytes(e.key)
+	e.key = r.ReadVarBytes()
```

Comment:
`(MaxKeyLength)`? Some safety checks should still be there. Especially given that we'll be transmitting/receiving these with proofs over the network, it's not just about local storage.

Reply:
Same as with the LeafNode, fixed by dropping the old commit.